src/backend/app/users/user_routes.py

```diff
@@ -66,3 +68,18 @@ async def get_user_roles(current_user: Annotated[DbUser, Depends(mapper)]):
     for role in UserRoleEnum:
         user_roles[role.name] = role.value
     return user_roles
+
+
+@router.delete("/{id}")
+async def delete_user_by_identifier(
+    user: Annotated[DbUser, Depends(get_user)],
+    current_user: Annotated[DbUser, Depends(super_admin)],
+    db: Annotated[Connection, Depends(db_conn)],
+):
+    """Delete a single user."""
+    log.info(
+        f"User {current_user.username} attempting deletion of user {user.username}"
+    )
+    await DbUser.delete(db, user.id)
+    log.info(f"User {user.id} deleted successfully.")
+    return Response(status_code=HTTPStatus.NO_CONTENT)
```

Comment:
I think it's best to return a response with a similar message as in the log instead of `NO_CONTENT`.